cura/MachineAction.py

```diff
@@ -114,3 +124,26 @@ def qmlPath(self) -> "QUrl":
     @pyqtSlot(result = QObject)
     def getDisplayItem(self) -> Optional["QObject"]:
         return self._createViewFromQML()
+
+    @pyqtSlot(result = bool)
+    def openAsDialog(self) -> bool:
+        """Whether this action will show a dialog.
+
+         If not, the action will directly run the function inside execute().
+
+        :return: Defaults to true to be in line with the old behaviour.
+        """
+
+        return self._open_as_dialog
+
+    @pyqtSlot(result = bool)
+    def isVisible(self) -> bool:
```

Comment:
Same as above, this is a property of the machine action, so feel that it should also be a property (and probably just a "visible" one). 

Since it's probably likely that you want this to be dynamic, I would also consider creating a setter that triggers the signal so that the interface is notified of the property changes.